src/server/scripts/EasternKingdoms/ShadowfangKeep/shadowfang_keep.cpp

```diff
@@ -205,6 +205,131 @@ class npc_arugal_voidwalker : public CreatureScript
+};
+
+enum ArugalTexts
+{
+    SAY_ARUGAL_AGGRO            = 1, // You, too, shall serve!
+    SAY_ARUGAL_TRANSFORM        = 2, // Release your rage!
+    SAY_ARUGAL_KILLED_UNIT      = 3  // Another falls!
+};
+
+enum ArugalEvents
+{
+    EVENT_ARUGAL_VOID_BOLT      = 1,
+    EVENT_ARUGAL_TELEPORT       = 2,
+    EVENT_ARUGAL_THUNDERSHOCK   = 3,
+    EVENT_ARUGAL_CURSE          = 4
+};
+
+std::array<uint32, 3> arugalTeleportSpells =
+{
+    SPELL_TELE_UPPER,
+    SPELL_TELE_STAIRS,
+    SPELL_TELE_SPAWN
+};
+
+class boss_archmage_arugal : public CreatureScript
+{
+public:
+    boss_archmage_arugal() : CreatureScript("boss_archmage_arugal") { }
+
+    struct boss_archmage_arugalAI : public BossAI
+    {
+        boss_archmage_arugalAI(Creature* creature) : BossAI(creature, BOSS_ARUGAL) { }
+
+        void Reset() override
+        {
+            _Reset();
+        }
+
+        void KilledUnit(Unit* who) override
+        {
+            if (who->GetTypeId() == TYPEID_PLAYER)
+                Talk(SAY_ARUGAL_KILLED_UNIT);
+        }
+
+        void SpellHitTarget(Unit* /*target*/, SpellInfo const* spell) override
+        {
+            if (spell->Id == SPELL_ARUGAL_CURSE)
+                Talk(SAY_ARUGAL_TRANSFORM);
+        }
+
+        void EnterCombat(Unit* /*who*/) override
+        {
+            _EnterCombat();
+            Talk(SAY_ARUGAL_AGGRO);
+            events.ScheduleEvent(EVENT_ARUGAL_CURSE, Seconds(7));
+            events.ScheduleEvent(EVENT_ARUGAL_TELEPORT, Seconds(15));
+            events.ScheduleEvent(EVENT_ARUGAL_VOID_BOLT, Seconds(1));
+            events.ScheduleEvent(EVENT_ARUGAL_THUNDERSHOCK, Seconds(10));
+        }
+
+        void AttackStart(Unit* who) override
+        {
+            if (!who)
+                return;
+
+            AttackStartCaster(who, 100.0f); // void bolt range is 100.f
+        }
+
+        void UpdateAI(uint32 diff) override
+        {
+            if (!UpdateVictim())
+                return;
+
+            if (me->HasUnitState(UNIT_STATE_CASTING))
+                return;
+
+            events.Update(diff);
+
+            while (auto eventId = events.ExecuteEvent())
+            {
+                switch (eventId)
+                {
+                    case EVENT_ARUGAL_CURSE:
+                        if (Unit* target = SelectTarget(SELECT_TARGET_RANDOM, 1, 30.0f, true))
+                            DoCast(target, SPELL_ARUGAL_CURSE);
+
+                        events.ScheduleEvent(eventId, Seconds(15));
+                        break;
+                    case EVENT_ARUGAL_TELEPORT:
+                    {
+                        auto spell = arugalTeleportSpells.front();
+                        DoCast(spell); 
+                        // rotate array, can be done with some kind of shuffle. Up to you to figure out what is best.
+                        std::rotate(arugalTeleportSpells.begin(), arugalTeleportSpells.begin() + 1, arugalTeleportSpells.end());
```

Comment:
Hmm. Point taken. Would it have made any difference if I had put "by
Riztazz" in caps + bold in the PR description & commit msg?

On Sat, Sep 10, 2016, 23:51 jackpoz <notifications@github.com> wrote:

> In src/server/scripts/EasternKingdoms/ShadowfangKeep/shadowfang_keep.cpp
> <https://github.com/TrinityCore/TrinityCore/pull/17922#discussion_r78284436>
> :
>
> > +            while (auto eventId = events.ExecuteEvent())
> > +            {
> > +                switch (eventId)
> > +                {
> > +                    case EVENT_ARUGAL_CURSE:
> > +                        if (Unit* target = SelectTarget(SELECT_TARGET_RANDOM, 1, 30.0f, true))
> > +                            DoCast(target, SPELL_ARUGAL_CURSE);
> > +
> > +                        events.ScheduleEvent(eventId, Seconds(15));
> > +                        break;
> > +                    case EVENT_ARUGAL_TELEPORT:
> > +                    {
> > +                        auto spell = arugalTeleportSpells.front();
> > +                        DoCast(spell);
> > +                        // rotate array, can be done with some kind of shuffle. Up to you to figure out what is best.
> > +                        std::rotate(arugalTeleportSpells.begin(), arugalTeleportSpells.begin() + 1, arugalTeleportSpells.end());
>
> It always confuses me to see PRs opened by other people than the author of
> the code, I thought tkrokli wrote all the changes
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/TrinityCore/TrinityCore/pull/17922/files/7ce72f0775af7a3e5786410e65de373478ebf689#r78284436>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AErSsKaRswJd5k4-4xJVKjumVAP_WOT-ks5qoyZYgaJpZM4J5U3J>
> .
>